.github/workflows/deploy-pages.yaml

```diff
@@ -1,11 +1,16 @@
 name: Deploy on GitHub Pages
 on:
-  push:
-    branches:
-      - development
+  workflow_dispatch:
+  pull_request:
+    types:
+      - labeled
 jobs:
   build-and-deploy:
+    if: ${{ github.event.action == 'labeled' && github.event.label.name == 'deploy' }}
```

Comment:
If you trigger this with a deploy label, you should also remove the label at the end of the run.

Reply:
The Dodona action uses a GitHub token for the remove label action. However, since I have no access to the repository settings, I cannot generate this access token myself. I will include it in this PR, can you add the secret?

Reply:
No, GITHUB_TOKEN is set automatically for the current user.